wallet/difficulty.go

```diff
@@ -170,20 +178,29 @@ func (w *Wallet) calcNextBlake256Diff(dbtx walletdb.ReadTx, header *wire.BlockHe
 			recentTime = olderTime
 		}
 
-		if i == nodesToTraverse {
+		if i == windowsToTraverse {
 			break // Exit for loop when we hit the end.
 		}
 
 		// Get the previous node while staying at the genesis block as needed.
 		// Query the header from the provided chain instead of database if
 		// present.  The parent of chain[0] is guaranteed to be in stored in the
 		// database.
-		if oldHeader.Height != 0 {
-			if len(chain) > 0 && int32(oldHeader.Height)-int32(chain[0].Header.Height) > 0 {
-				oldHeader = chain[oldHeader.Height-chain[0].Header.Height-1].Header
+		if int64(oldHeight) > params.WorkDiffWindowSize {
+			oldHeight -= uint32(params.WorkDiffWindowSize)
+		} else {
+			oldHeight = 0
+		}
+		if oldHeight != 0 {
+			if len(chain) > 0 && int32(oldHeight)-int32(chain[0].Header.Height) >= 0 {
+				idx := oldHeight - chain[0].Header.Height
```

Comment:
the previous code had an additional -1 in this index calculation.  was that incorrect, or is it just missing here?

Reply:
The logic is slightly different between the old and new versions.

The old version was backtracking one block at a time, so that `-1` in this branch was effectively the iteration (i.e. "pick previous block").

In the new version, the backtracking is done on the prior code block, by a whole window (the `-= params.WorkDiffWindowSize`), so the -1 is unnecessary here (because we already established what the previous block index must be).